extensions/amp-access/0.1/amp-login-done-dialog.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {listen} from '../../../src/event-helper';
+
+
+/**
+ * @private Visible for testing.
+ */
+export class LoginDoneDialog {
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const {!Window} */
+    this.win = win;
+  }
+
+  /**
+   * Runs all the steps in processing. First, the dialog tries to postback
+   * results to the opener window via messaging. If opener is not available
+   * or if it timeouts, the dialog switches to error mode where the "close"
+   * button will be available.
+   */
+  start() {
+    this.setStyles_();
+    this.postback_().then(
+        this.postbackSuccess_.bind(this),
+        this.postbackError_.bind(this));
+  }
+
+  /**
+   * Sets necessary CSS styles to select the language. See `buildStyles_` for
+   * more details.
+   * @private
+   */
+  setStyles_() {
+    const doc = this.win.document;
+    const style = doc.createElement('style');
+    style.textContent = this.buildStyles_();
+    doc.head.appendChild(style);
+  }
+
+  /**
+   * The language is selected based on `navigate.language` by setting CSS of
+   * the following form:
+   * ```
+   * [lang="fr"], [lang="fr-FR"] {display: block}
+   * ```
+   * @return {string}
+   * @private
+   */
+  buildStyles_() {
+    const nav = this.win.navigator;
+    const lang = nav.language || nav.userLanguage || 'en-US';
```

Comment:
Ok. I'll add the attribute and default to `navigator.language` when not specified.